tools/nblIntrinicGenerator/example output/exampleIntrisincsCore.hlsl

```diff
@@ -0,0 +1,185 @@
+
+// Copyright (C) 2024 - DevSH Graphics Programming Sp. z O.O.
+// This autogenerated file is part of the "Nabla Engine".
+// This file was created using Spirv HLSL intrinsics generator
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+
+#ifndef _EXAMPLE_OUTPUT_EXAMPLE_INTRISINCS_CORE_HLSL_INCLUDED_
+#define _EXAMPLE_OUTPUT_EXAMPLE_INTRISINCS_CORE_HLSL_INCLUDED_
+
+
+#ifdef __HLSL_VERSION
+#include "spirv/unified1/spirv.hpp"
+#include "spirv/unified1/GLSL.std.450.h"
+#endif
+
+
+namespace nbl
+{
+namespace hlsl
+{
+#ifdef __HLSL_VERSION
+namespace spirv
+{
+// Built-ins
+
+[[vk::ext_builtin_input(spv::BuiltInHelperInvocation)]]
+static const bool HelperInvocation;
+
+[[vk::ext_builtin_input(spv::BuiltInPosition)]]
+static float32_t4 Position;
+
+[[vk::ext_builtin_input(spv::BuiltInVertexIndex)]]
+static const uint32_t VertexIndex;
+
+[[vk::ext_builtin_input(spv::BuiltInInstanceIndex)]]
+static const uint32_t InstanceIndex;
+
+[[vk::ext_builtin_input(spv::BuiltInNumWorkgroups)]]
+static const uint32_t NumWorkgroups;
+
+[[vk::ext_builtin_input(spv::BuiltInWorkgroupId)]]
+static const uint32_t WorkgroupId;
+
+[[vk::ext_builtin_input(spv::BuiltInLocalInvocationId)]]
+static const uint32_t LocalInvocationId;
+
+[[vk::ext_builtin_input(spv::BuiltInGlobalInvocationId)]]
+static const uint32_t GlobalInvocationId;
+
+[[vk::ext_builtin_input(spv::BuiltInLocalInvocationIndex)]]
+static const uint32_t LocalInvocationIndex;
+
+
+
+// Class 'Atomic'
+// OpAtomicLoad
+template<typename T, typename P>
+[[vk::ext_instruction(spv::OpAtomicLoad)]]
+T AtomicLoad([[vk::ext_reference]] P pointer, uint32_t memory, uint32_t semantics);
+
+// OpAtomicStore
+template<typename T, typename P>
+[[vk::ext_instruction(spv::OpAtomicStore)]]
+void AtomicStore([[vk::ext_reference]] P pointer, uint32_t memory, uint32_t semantics, T value);
+
+// OpAtomicExchange
+template<typename T, typename P>
+[[vk::ext_instruction(spv::OpAtomicExchange)]]
+T AtomicExchange([[vk::ext_reference]] P pointer, uint32_t memory, uint32_t semantics, T value);
+
+// OpAtomicCompareExchange
+template<typename T, typename P>
+[[vk::ext_instruction(spv::OpAtomicCompareExchange)]]
+T AtomicCompareExchange([[vk::ext_reference]] P pointer, uint32_t memory, uint32_t equal, uint32_t unequal, T value, T comparator);
+
+// OpAtomicCompareExchangeWeak
+template<typename T, typename P>
+[[vk::ext_instruction(spv::OpAtomicCompareExchangeWeak)]]
+T AtomicCompareExchangeWeak([[vk::ext_reference]] P pointer, uint32_t memory, uint32_t equal, uint32_t unequal, T value, T comparator);
```

Comment:
I think you need to add a "filter" to catch C++ reserved keywords (case sensitive) like and,or,equal,restrict,register and "tweak" then to be prefixed with a single or double `_`

Reply:
ill do this for all of these, it seems like its up to date 
https://www.informit.com/articles/article.aspx?p=3128876&seqNum=7